statsmodels/sandbox/stats/diagnostic.py

```diff
@@ -283,17 +283,14 @@ def acorr_ljungbox(x, lags=None, boxpierce=False):
     x = np.asarray(x)
     nobs = x.shape[0]
     if lags is None:
-        lags = lrange(1,41)  #TODO: check default; SS: changed to 40
+        lags = lrange(1, min((nobs // 2 - 2), 40) + 1)
```

Comment:
extra possible cleanup
lrange is not needed, we could use np.arange, because below it is converted with np.asarray